scripts/tests.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+# NOTE
+# If some crate fails, you can continue the testing after fixing it as follows:
+# ./scripts/tests.sh <failed-crate>
+
+start_from=$1
+
+all_crates=$(
+    cargo workspaces list
+)
+
+cargo_action() {
+    action=$1
+    package=$2
+    features=$3
+
+    echo -e "$testing_prompt cargo $action -p $package $features"
+    cargo $action -p $package $features
+
+    if [[ $1 -ne 0 ]]
+    then
+        echo "Aborting!"
+        exit 1
+    fi
+}
+
+ESC="\033"
+testing_prompt="$ESC[1;36m     Testing$ESC[0m"
+go=0
+
+if [[ -z "$start_from" ]]; then
+    go=1
+fi
+
+cargo install cargo-workspaces
```

Comment:
Can we have this as part of a `setup` script? Having this here means that every time we want to run this (locally) we will have cargo checking for updates and trying to update this crate, which will slow this process even further. wdyt?

Reply:
Sure! Make sense. What's the `setup` script you're referring?

Reply:
It doesn't quite exist atm, but we could create one or adapt the current install-toolchain to be more general-purpose and install tools we need for development

Reply:
Ok, I understand. By now, f you agree, I can print a message if the command is not found instead of fetching every time if it exists. I do not want to add things into `install_toolchain.sh` because it affects all CI job time/space. Only the required job should fetch its required tools, I think.

Reply:
We can prepare a good script in another PR, with taplo, subalfred, cargo-workspaces, ...

Reply:
Yeah that's fine :)